### DISCUSSION TITLE
class ServiceConnectionProvider is not exported.

### DISCUSSION THREAD

**Discussion by Camork** [eclipse-theia/theia#13669] [General] [answered]
Hello community

I am trying the Theia demos from the generator, and mentioned that the class `WebSocketConnectionProvider` is marked deprecated by `ServiceConnectionProvider` but `ServiceConnectionProvider` is not exported. Is this a mistake?

**Comment by msujew** [comment, marked as answer]
Hey @Camork,

you can import it from its full CJS path: `@theia/core/lib/browser/messaging/service-connection-provider`.
